tests/testthat/_snaps/build-articles.md

```diff
@@ -30,6 +30,43 @@
       Reading vignettes/html-vignette.Rmd
       Writing `articles/html-vignette.html`
 
+# bad width gives nice error
+
+    Code
+      build_rmarkdown_format(pkg, "article")
+    Condition
+      Error in `build_rmarkdown_format()`:
+      ! code.width must be a whole number, not the string "abc".
+      i Edit _pkgdown.yml to fix the problem.
+
+# validates articles yaml
+
+    Code
+      data_articles_index_(1)
+    Condition
+      Error in `data_articles_index_()`:
+      ! articles must be a list, not the number 1.
+      i Edit _pkgdown.yml to fix the problem.
+    Code
+      data_articles_index_(list(1))
+    Condition
+      Error in `data_articles_index_()`:
+      ! articles[1] must be a list, not the number 1.
```

Comment:
Do you think it's reasonably clear that I'm talking about the first component of `articles` here?

Reply:
Why not write it out and put articles as code

Component 1 of `articles` must be a list, not the number 1.

In any case it's helpful that the last line of the error refers to the config.

Reply:
I'm not convinced that spelling it out in words will help. I think the mapping between the R data structures and the yaml structure is likely to be confusing, but I don't think that there's much better we can do until RStudio supports feedback directly on the yaml file (which I think is likely to come by the end of the year).

I do wonder if it would be better to put the file name closer, e.g. `_pkgdown.yml: articles[1] must be a list...`. (Making `_pkgdown.yml` a link so we can remove the edit bullet at the end).